addons/web/static/tests/views/form_tests.js

```diff
@@ -2254,6 +2255,66 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test("modal should not use the widget footer elements", async function (assert) {
+        assert.expect(2);
+
+        var MyWidget = basicFields.FieldChar.extend({
+            className: 'oe_form_field d-flex',
+            _renderEdit: function () {
+                var res = this._super();
+                const $el = $(
+                    `<section class="oe_form_field">
+                        <header>title</header>
+                        <article>content ` + this.$el[0].outerHTML + `</article>
+                        <footer><button>button</button></footer>
+                    </section>`);
+                this.$el.replaceWith($el);
+                this.$el = $el;
+                return res;
+            },
+        });
```

Comment:
You can simplify this a lot. Just define a widget with a `<footer>` tag. And with my suggestion above, you don't need the oe_form_field classname anymore.

Reply:
No because in my case the class is not on the footer but on a parent. This test corresponds to the current issue.